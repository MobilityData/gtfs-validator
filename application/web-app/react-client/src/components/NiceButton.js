/*
 *  Copyright (c) 2020. MobilityData IO.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import React from 'react'
import {Button} from 'semantic-ui-react'

/**
 * Customized button
 * @param id {string} id of the DOM element
 * @param method {function} method to be triggered by `onclick` event
 * @param description {string} text to be displayed inside the button
 * @returns {JSX.Element} the DOM element to be rendered
 * @constructor
 */
const NiceButton = ({id, method, description}) => <Button fluid id={id} onClick={method}>{description}</Button>

export default NiceButton
