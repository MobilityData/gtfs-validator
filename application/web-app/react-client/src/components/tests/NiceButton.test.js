/*
 *  Copyright (c) 2020. MobilityData IO.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import React from 'react';
import NiceButton from "../NiceButton";
import renderer from 'react-test-renderer';

describe('NiceButton', () => {
    let mockOnClick;
    beforeAll(() => {
        mockOnClick = jest.fn();
    });

    it('should be defined', () => {
        expect(NiceButton).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = renderer.create(
            <NiceButton id='id value' onClick={mockOnClick} description="description value"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call function when button is clicked', () => {
        const tree = renderer.create(
            <NiceButton id='id value' onClick={mockOnClick} description="description value"/>
        );
        tree.root.props.onClick();
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});
