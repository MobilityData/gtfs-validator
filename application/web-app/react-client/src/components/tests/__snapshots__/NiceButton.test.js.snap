// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NiceButton should render correctly 1`] = `
<button
  className="ui fluid button"
  id="id value"
  onClick={[Function]}
>
  description value
</button>
`;
